Added syscheck fields specification and example rules (#3913)
wazuh/wazuh-documentation: source/user-manual/capabilities/file-integrity/fim-fields-rule-mapping.rst

.. Copyright (C) 2021 Wazuh, Inc.

> Fim fields rule mapping

Reviewer: ```suggestion
FIM fields rule mapping
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

=======================

This guide aims to help to understand how FIM fields can be mapped into rules.

FIM - Alerts : fields correspondence
------------------------------------

> The following table establishes a correspondence between the syscheck fields and their homologs in the alert generation.

Reviewer: ```suggestion
The following table establishes a correspondence between the decoded FIM fields and their counterpart in rules.
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  **Syscheck field**  |  **Alert field**            | **Field description**                                           |

> +----------------------+-----------------------------+-----------------------------------------------------------------+

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

> |  file                |  path                       |  File path in the current alert                                 |

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  size                |  size_after                 |  File size in the current alert                                 |

Reviewer: ```suggestion
|  size                |  size_after                 |  File size in the current event                                 |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  hard_links          |  hard_links                 |  List of hard links of the file                                 |
+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  mode                |  mode                       |  Syscheck scan mode                                             |

Reviewer: ```suggestion
|  mode                |  mode                       |  FIM event mode                                                 |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  perm                |  perm_after,win_perm_after  |  File permissions                                               |
+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  uid                 |  uid_after                  |  User ID of the process                                         |

Reviewer: ```suggestion
|  uid                 |  uid_after                  |  User ID of the owner of the file                                         |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  gid                 |  gid_after                  |  Group ID of the process                                        |

Reviewer: ```suggestion
|  gid                 |  gid_after                  |  Group ID of the group that shares ownership of the file               |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  md5                 |  md5_after                  |  md5 hash of the file in the current alert (after changes)      |

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  sha1                |  sha1_after                 |  sha1 hash of the file in the current alert (after changes)     |

Reviewer: ```suggestion
|  sha1                |  sha1_after                 |  sha1 hash of the file in the current event (after changes)     |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  uname               |  uname_after                |  User name of the process                                       |

Reviewer: ```suggestion
|  uname               |  uname_after                |  User name of the owner of the file                                       |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  gname               |  gname_after                |  Group name of the process                                      |

Reviewer: ```suggestion
|  gname               |  gname_after                |  Group name of the group that shares ownership of the file                                      |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  mtime               |  mtime_after                |  Timestamp of the file changes                                  |
+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  inode               |  inode_after                |  Inode of the file in the current alert                         |

Reviewer: ```suggestion
|  inode               |  inode_after                |  Inode of the file in the current event                         |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  sha256              |  sha256_after               |  sha256 hash of the file in the current alert (after changes)   |

Reviewer: ```suggestion
|  sha256              |  sha256_after               |  sha256 hash of the file in the current event (after changes)   |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/f3e4eedc32cbd94b8cf88e3709d12d722b0e9dee)

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  changed_content     |  diff                       |  Reported changes on the file of the current alert              |

Reviewer: ```suggestion
|  changed_content     |  diff                       |  Reported changes on the file of the current event              |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/f3e4eedc32cbd94b8cf88e3709d12d722b0e9dee)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  changed_fields      |  changed_attributes         |  Changed fields in the file (permissions, content, etc...)      |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  win_attributes      |  attrs_after                |  File attributtes (hidden, read-only, etc...)                   |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  tag                 |  tag                        |  Custom tags to be added to one specific alert                  |
+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  symbolic_path       |                             |                                                                 |

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  parent_name         |  audit.process.parent_name  |  (Audit) parent process name                                    |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  cwd                 |  audit.process.cwd          |  (Audit) current work directory                                 |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  parent_cwd          |  audit.process.parent_cwd   |  (Audit) parent process's current work directory                |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  audit_uid           |  audit.login_user.id        |  (Audit) User login ID used by Audit                            |
+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  audit_name          |  audit.login_user.name      |  (Audit) User name used by Audit                                |

Reviewer: The audit fields give information about the user/process that caused changes that ultimately resulted in the current event. We should rephrase these fields so that we explicitly say the information is about the audit event that caused the FIM event to be triggered.

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/d5ce0fcadd5abf8bbb40c82c8ccbba1b333c2572)

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/6f538ba0f29e9eb74ddfc9e363f4de470e21bb84)

+----------------------+-----------------------------+-----------------------------------------------------------------+

> |  arch                |  arch                       |  Registry architecture (32 or 64)                               |

Reviewer: ```suggestion
|  arch                |  arch                       |  Registry architecture (32 or 64 bits)                           |
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/f3e4eedc32cbd94b8cf88e3709d12d722b0e9dee)

+----------------------+-----------------------------+-----------------------------------------------------------------+
|  value_name          |  value_name                 |  Registry value name                                         |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  value_type          |  value_type                 |  Registry value type                                         |
+----------------------+-----------------------------+-----------------------------------------------------------------+
|  entry_type          |  entry_type                 |  Registry entry type                                          |
+----------------------+-----------------------------+-----------------------------------------------------------------+

Rule mapping examples
---------------------

> The following example rules aim to show how to apply syscheck fields to correctly extract information from alerts.

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/f3e4eedc32cbd94b8cf88e3709d12d722b0e9dee)

.. Copyright (C) 2021 Wazuh, Inc.

FIM fields rule mapping
=======================

This guide aims to help to understand how FIM fields can be mapped into rules.

> FIM - Alerts : fields correspondence

Reviewer 2: ```suggestion
FIM - Alerts: fields correspondence
```

Author: Done in [commit](https://github.com/wazuh/wazuh-documentation/pull/3913/commits/1096a0103d3f508c9dde14df124066b17050626c)
